source/mechanics/force_contribution_lib.cc

```diff
@@ -387,6 +387,155 @@ namespace fdl
+  {
+    return true;
+  }
+
+  template <int dim, int spacedim, typename Number>
+  void
+  PericardiumModelForce<dim, spacedim, Number>::compute_boundary_force(
+    const double /*time*/,
+    const MechanicsValues<dim, spacedim>                              &m_values,
+    const typename Triangulation<dim, spacedim>::active_face_iterator &face,
+    ArrayView<Tensor<1, spacedim, Number>> &forces) const
+  {
+    if (this->boundary_ids.size() > 0 &&
+        !std::binary_search(this->boundary_ids.begin(),
+                            this->boundary_ids.end(),
+                            face->boundary_id()))
+      {
+        // the user specified a subset of boundary ids and we currently don't
+        // match - fill with zeros
+        for (auto &force : forces)
+          force = 0.0;
+      }
+    else
+      {
+        const FEValuesBase<dim, spacedim> &fe_values = m_values.get_fe_values();  // fe_value
+
+        const auto                         cell      = fe_values.get_cell();      // active cell
+
+        const auto                         dof_cell =                             // dof handler for the active cell
+          typename DoFHandler<dim, spacedim>::active_cell_iterator(
+            &this->dof_handler->get_triangulation(),
+            cell->level(),
+            cell->index(),
+            &*this->dof_handler);
+
+        this->scratch_cell_dofs.resize(fe_values.dofs_per_cell);                  // resize to number of cell dofs 
+
+        dof_cell->get_dof_indices(this->scratch_cell_dofs);                       // populate with dof ids for the active cell 
+
+        this->scratch_dof_values.resize(fe_values.dofs_per_cell);                 // resize to the number of cell dofs
+
+        this->scratch_qp_values.resize(fe_values.n_quadrature_points);            // resize to the number of cell dofs 
+
+        auto &extractor = fe_values[FEValuesExtractors::Vector(0)];               // FEValuesViews object 
+
+        for (unsigned int i = 0; i < this->scratch_cell_dofs.size(); ++i)         // loop over active cells 
+          this->scratch_dof_values[i] = this->spring_constant *                   // compute spring force at dofs
+            (this->reference_position[this->scratch_cell_dofs[i]] -
+            (*this->current_position)[this->scratch_cell_dofs[i]]);
+
+        extractor.get_function_values_from_local_dof_values(                      // interpolate spring forces at qps 
+          this->scratch_dof_values, this->scratch_qp_values);                     
+
+        for (unsigned int i = 0; i < this->scratch_qp_values.size(); ++i)         // add velocity damping
+          this->scratch_qp_values[i] = fe_values.normal_vector(i) *             // does this need to use FEFaceValues 
+            (this->scratch_qp_values[i] -  
+             this->damping_constant * m_values.get_velocity_values()[i]) *        // velocity values is at the whole cell QPs?
+            fe_values.normal_vector(i); 
```

Comment:
These should be the deformed normal vectors provided by `MechanicsValues`, not the reference normal vectors provided by `FEValues`.